modules/drivers/sensors/hx711/modHX711c.c

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2023 by Thorsten von Eicken.
+
+#include "xsPlatform.h"
+#include "xsmc.h"
+#include "modGPIO.h"
+#include "mc.xs.h" // for xsID_* constants
+
+#define xsmcVar(x) xsVar(x)
+
+typedef struct {
+  modGPIOConfigurationRecord clk;
+  modGPIOConfigurationRecord din;
+  int gain;
+} hx711_data;
+
+void xs_HX711_init(xsMachine *the) {
+  if (xsmcArgc != 3) xsUnknownError("invalid arguments");
```

Comment:
This check isn't strictly necessary. Using xsArg on an argument that doesn't exist will throw. All this guarantees is that there aren't more than 3 arguments. No harm in the check, but it is common for JavaScript functions to simply ignore extra arguments.